search/django/weedid/tasks.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import, unicode_literals
+from celery import shared_task
+from weedcoco.repo.deposit import deposit
+from weedcoco.index.indexing import ElasticSearchIndex
+from weedcoco.index.thumbnailing import thumbnailing
+from weedid.models import Dataset
+from core.settings import THUMBNAILS_DIR, REPOSITORY_DIR, DOWNLOAD_DIR
+from pathlib import Path
+
+
+@shared_task
+def upload_task(weedcoco_path, image_dir, upload_id):
+    upload_entity = Dataset.objects.get(upload_id=upload_id)
+    upload_entity.upload_status = "P"
+    upload_entity.upload_status_details = ""
+    upload_entity.save()
+    try:
+        new_weedcoco_path = deposit(
+            Path(weedcoco_path),
+            Path(image_dir),
+            Path(REPOSITORY_DIR),
+            Path(DOWNLOAD_DIR),
+            upload_id,
+        )
+    except Exception as e:
+        upload_entity.upload_status = "F"
+        upload_entity.upload_status_details = str(e)
```

Comment:
We should log the full traceback, either in the DB or in server logs (or both), although it does not need to be user-visible in `status_details`. See `traceback.format_exc` for what you need to do so.